### PR TITLE
feat: 7109 scaffold new HubGroup entity

### PR DESCRIPTION
**Pull request by vivzhang**
1. Description:

Scaffold a new HubGroup entity by adding:
- new `IHubGroup` interface
- new `HubGroups` with GRUD and other methods 
- new `HubGroup` class

1. Instructions for testing:

1. Closes Issues: [7109](https://zentopia.esri.com/workspaces/collaboration-sprint-board-614f9bfc0946c40011ef574e/issues/dc/hub/7109)

1. [ ] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

**Review by dbouwman — CHANGES_REQUESTED** [review]
This is a solid start - we need to review a number of the `IHubGroup` properties and how they are mapped to/from the `IGroup`.

File: packages/common/src/core/types/IHubGroup.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+export interface IHubGroup extends IHubEntityBase, IWithPermissions {
+  access: "private" | "org" | "public";
+  autoJoin?: boolean;
+  description?: string;
+  isInvitationOnly?: boolean;
+  isDiscussable?: boolean;
+  isReadOnly?: boolean;
+  isViewOnly?: boolean;
+  membershipAccess?: string;
```
[the remark this answers is not in the document]
Comment (vivzhang) [reply packages/common/src/core/types/IHubGroup.ts:17]: According to [`this confluence`](https://confluencewikidev.esri.com/pages/viewpage.action?pageId=105316739),  "Privileged users will be able to specify who can, thus who cannot, join any group in their organization." If this prop is passed in when a group is created, the default value will be "org".

File: packages/common/src/groups/HubGroup.ts [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+   */
+  removePermissionPolicy(permission: Permission, id: string): void {
+    this.entity.permissions = removePermissionPolicy(
+      this.entity.permissions,
+      permission,
+      id
+    );
+  }
+
+  /**
```
Comment (dbouwman) [inline packages/common/src/groups/HubGroup.ts:236]: We can just remove this

**Review by dbouwman — APPROVED** [review]
Only one small ask - please add some permissions into the e2e - just to verify that the permissions are stored in `.properties.permissions` and then read back out. I'm like 99% sure this will be fine, but it's a good practice to verify :)

**Pull request merged**